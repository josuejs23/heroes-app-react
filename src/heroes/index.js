export * from './pages'
export * from './routes/HereoesRoutes'