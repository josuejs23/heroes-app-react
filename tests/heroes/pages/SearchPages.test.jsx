import {render, screen, fireEvent} from '@testing-library/react';
import { MemoryRouter, useNavigate } from 'react-router-dom';
import {AuthContext} from '../../../src/auth';
import {SearchPage} from '../../../src/heroes';

const mockedUseNavigate = jest.fn();
jest.mock('react-router-dom', ()=>({
    ...jest.requireActual('react-router-dom'),
    useNavigate:()=>mockedUseNavigate
}))

describe('Test on <SearchPage/>', () => { 
    test('Should make match with the snapshot', ()=>{

        const authContextValue = {
            logged : true,
            user:{
                name:'Josue Zorrilla',
                id:2302
            }
        }
        const {container} = render(
            <MemoryRouter>
                <SearchPage/>
            </MemoryRouter>
        )
        
        expect(container).toMatchSnapshot();
    })

    test('Should show batman', ()=>{

        
        render(
            <MemoryRouter initialEntries={['/search?q=batman']}>
                <SearchPage/>
            </MemoryRouter>
        )
        // screen.debug();
        const img = screen.getByRole('img');
        const divSearchHero = screen.getByText('Search a hero');
        const divNoHero = screen.getByText(`There's is no hero: batman`);
        expect(img.src).toContain("/assets/heroes/dc-batman.jpg");
        expect(screen.getByRole('heading',{level:5}).innerHTML).toContain('Batman')
        expect(divSearchHero.style._values).toEqual({"display": "none"})
        expect(divNoHero.style._values).toEqual({"display": "none"})
    })

    test('Should show a error if cant find the heroe', ()=>{
        render(
            <MemoryRouter initialEntries={['/search?q=xxxxxx']}>
                <SearchPage/>
            </MemoryRouter>
        )
        expect(screen.getByText(`There's is no hero: xxxxxx`)).toBeTruthy();
    })

    test('Should navigate to the hero page', ()=>{
        render(
            <MemoryRouter initialEntries={['/search']}>
                <SearchPage/>
            </MemoryRouter>
        )
        const textBox = screen.getByRole('textbox');
        fireEvent.change(textBox,{target : {name:'searchText',value:'batman'}})
        const form = screen.getByRole('form');
        fireEvent.submit(form);
        // screen.debug();
        expect(mockedUseNavigate).toBeCalledWith('?q=batman');
    })
 })