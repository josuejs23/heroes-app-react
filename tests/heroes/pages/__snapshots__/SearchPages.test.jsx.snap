// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <SearchPage/> Should make match with the snapshot 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form
        aria-label="form"
      >
        <input
          autocomplete="off"
          class="form-control"
          id=""
          name="searchText"
          placeholder="Search hero..."
          type="text"
          value=""
        />
        <button
          class="btn btn-primary mt-1"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results:
      </h4>
      <hr />
      <div
        class="alert alert-danger"
        style="display: none;"
      >
        There's is no hero: 
      </div>
      <div
        class="alert alert-primary"
      >
        Search a hero
      </div>
    </div>
  </div>
</div>
`;
